### PR TITLE
Add JSON tags to Limits object for JSON serialization

### PR DESCRIPTION
**Pull request by 56quarters**
This allows Limits objects to be included in JSON HTTP APIs by
downstream Cortex consumers.

Command run to update the tags:

```
perl -pi -e 's/`yaml:"([\w,\-]+)"/`yaml:"$1" json:"$1"/' pkg/util/validation/limits.go
```

Signed-off-by: Nick Pillitteri <nick.pillitteri@grafana.com>

**Checklist**
- [x] Tests updated

### CODE REVIEW

**Review by jtlisi — APPROVED** [review]
LGTM

File: pkg/util/validation/limits_test.go [hunk 1]
```diff
@@ -110,6 +112,43 @@ func TestLimitsLoadingFromYaml(t *testing.T) {
...
+
+	l := Limits{}
+	err := json.Unmarshal([]byte(inp), &l)
+	require.NoError(t, err)
+
+	assert.Equal(t, 0.5, l.IngestionRate, "from json")
+	assert.Equal(t, 100, l.MaxLabelNameLength, "from defaults")
+}
+
+func TestLimitsTagsYamlMatchJson(t *testing.T) {
```
Comment (jtlisi) [inline pkg/util/validation/limits_test.go:130]: Great idea for a test

Comment (pracucci) [reply pkg/util/validation/limits_test.go:130]: Agree, that's pretty cool. Good job!

Comment (pracucci) [reply pkg/util/validation/limits_test.go:130]: This test will pass anyway if a file has only YAML and another field has only JSON. Have you considered checking that the yaml and json tag of each field exactly match (the content I mean)?

**Review by pracucci — APPROVED** [review]
LGTM! (modulo a nit about the test)

File: pkg/util/validation/limits_test.go (hunk 1 above)
Comment (56quarters) [reply pkg/util/validation/limits_test.go:130]: Good point. I'll adjust the test to fix that :+1:

File: pkg/util/validation/limits_test.go [hunk 2]
```diff
@@ -110,6 +112,42 @@ func TestLimitsLoadingFromYaml(t *testing.T) {
...
+	n := limits.NumField()
+	var mismatch []string
+
+	for i := 0; i < n; i++ {
+		field := limits.Field(i)
+
+		// Note that we aren't requiring YAML and JSON tags to match, just that
+		// they either both exist or both don't exist.
+		hasYamlTag := field.Tag.Get("yaml") != ""
+		hasJsonTag := field.Tag.Get("json") != ""
```
Comment (pracucci) [inline pkg/util/validation/limits_test.go:141]: We have a picky linter :)

```suggestion
		hasJSONTag := field.Tag.Get("json") != ""
```

**Pull request merged**